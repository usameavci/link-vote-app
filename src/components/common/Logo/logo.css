.common-logo {
  height: 38px;
  display: block;
  filter: grayscale(1);
}
