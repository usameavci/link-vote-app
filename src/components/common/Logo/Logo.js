import React from "react";
import { Link } from "react-router-dom";

import LogoUrl from "../../../assets/images/logo.jpg";

import "./logo.css";

const Logo = () => {
  return (
    <Link data-testid="common-logo" to="/">
      <img src={LogoUrl} alt="Hepsiburada" className="common-logo" />
    </Link>
  );
};

export default Logo;
