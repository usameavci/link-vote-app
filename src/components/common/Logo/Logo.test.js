import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import Logo from "./Logo";

describe("Logo component tests", () => {
  it("renders without crashing", async () => {
    const component = (
      <Router>
        <Logo />
      </Router>
    );

    const { getByTestId } = render(component);

    expect(getByTestId("common-logo")).toBeInTheDocument();
  });

  it("matches snapshots", () => {
    const component = (
      <Router>
        <Logo />
      </Router>
    );

    const tree = renderer.create(component);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
