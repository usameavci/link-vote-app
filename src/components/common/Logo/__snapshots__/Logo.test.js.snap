// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component tests matches snapshots 1`] = `
<a
  data-testid="common-logo"
  href="/"
  onClick={[Function]}
>
  <img
    alt="Hepsiburada"
    className="common-logo"
    src="logo.jpg"
  />
</a>
`;
